File: Leetcode/JS/Leetcode507-完美数.js
var checkPerfectNumber = function(num) {
    let ans = 0;
    for(let i = 1;i <= num / i;i++){
        if(num % i == 0) {
            if(i < num) ans += i;
            if(i != Math.floor(num / i) && Math.floor(num / i) < num) ans += Math.floor(num / i);
        }
    }
    return ans == num;
};

File: Leetcode/sbgo/Leetcode507-完美数.go
package sbgo

func checkPerfectNumber(num int) bool {
	ans := 0
	for i := 1; i <= num/i; i++ {
		if num%i == 0 {
			if i < num {
				ans += i
			}
			if num/i < num && i != num/i {
				ans += num / i
			}
		}
	}
	return ans == num
}
